code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -713,7 +772,14 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 					if(last_open <= world.time - 20) //Spam limiter.
 						last_open = world.time
 						user.visible_message(SPAN_DANGER("After a few attempts, <b>[user]</b> manages to light \the [src], they however burn their finger in the process."), range = 3)
-			set_light(2, 1, l_color = LIGHT_COLOR_LAVA)
+			if(istype(src, /obj/item/flame/lighter/zippo/blue))
+				set_light(2, 1, l_color = LIGHT_COLOR_BLUE)
+			else if(istype(src, /obj/item/flame/lighter/zippo/purple))
+				set_light(2, 1, l_color = LIGHT_COLOR_VIOLET)
+			else if(istype(src, /obj/item/flame/lighter/zippo/himeo))
+				set_light(2, 2, l_color = LIGHT_COLOR_LAVA)
+			else
+				set_light(2, 1, l_color = LIGHT_COLOR_LAVA)
```

Comment:
`istype(src, ...)` is something that should never be done. Make this a function and call it here, then override it on each zippo.

Reply:
All done, chief.